eZ/Publish/Core/IO/IOMetadataHandler/LegacyDFSCluster.php

```diff
@@ -242,7 +242,7 @@ public function getMimeType($spiBinaryFileId)
     public function deleteDirectory($spiPath)
     {
         $stmt = $this->db->prepare('DELETE FROM ezdfsfile WHERE name LIKE ?');
-        $stmt->bindValue(1, rtrim($spiPath, '/') . '/%');
+        $stmt->bindValue(1, $this->prepareLikeString($this->addPrefix(rtrim($spiPath, '/'))) . '/%');
```

Comment:
This is DBMS-specific :upside_down_face: 
E.g. Doesn't work as expected in SQLite.

There's another SQL keyword: `ESCAPE` which is used to set escape character for `LIKE`. Seems to be working for SQLite, MySQL, PostgreSQL and Oracle at least.

While at this, could you improve this part to use `Doctrine\DBAL\Query\QueryBuilder`? Something like:
```php
$query = $this->db->createQueryBuilder();
$query
    ->delete('ezdfsfile')
    ->where('name LIKE :spiPath ESCAPE :esc')
    ->setParameter(':esc', '\\')
    ->setParameter(
        ':spiPath',
        addcslashes($this->addPrefix(rtrim($spiPath, '/')), '%_') . '/%'
    );
$query->execute();
```
Note that:
- I'm intentionally quoting escape sequence as this also might be DBMS-specific (don't think it is though, but you never know... :grinning:).
- I'm explicitly using here `addcslashes` instead of wrapper method so it's clear why `ESCAPE` is needed.

Note to self: somewhere in the next Millennium I should introduce here Gateway.

Reply:
@alongosz More standard way will be better, but be aware that kernel already escapes wildcards in LIKE using C slashes: 

https://github.com/adamwojs/ezpublish-kernel/blob/master/eZ/Publish/Core/Search/Legacy/Content/Common/Gateway/CriterionHandler/FieldValue/Handler.php#L135  

(that's why I've approved this PR)

Reply:
> More standard way will be better, but be aware that kernel already escapes wildcards in LIKE using C slashes

@adamwojs then probably one more thing not extensively covered in integration tests ;)

`LIKE '\_something'` simply does not work across every DBMS (SQLite might be the only one though :grinning: ).

Reply:
that's why I did it in this way originally )

Reply:
@alongosz @adamwojs 
Let me know if you would like to see original version with `prepareLikeString` in this PR and work on "ESCAPE" issue separately.

Reply:
> Let me know if you would like to see original version with prepareLikeString in this PR and work on "ESCAPE" issue separately.

No, this is fine now :)